### PR TITLE
feat(react-tabs): add circular tab appearance variants

### PR DESCRIPTION
**Pull request by dmytrokirpa**
## New Behavior

Adds a new `appearance="subtle-circular"` and `appearance="filled-circular"` variants to `TabList` and `Tab` components.

https://github.com/user-attachments/assets/3c3160f3-131c-4f58-aa36-6815f286e7f4

Figma - https://www.figma.com/design/dK5AnDvvnSTWV9lduQWeDk/TabList?node-id=11485-46208&t=mGEm2A0P5LNHTYUs-1

## Related Issue(s)

- Implements #32951

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">40</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">29</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Avatar-mount_base.html">639</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Avatar-mount.html">669</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Button-mount_base.html">315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Button-mount.html">314</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Field-mount_base.html">1207</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Field-mount.html">1177</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProvider-mount_base.html">716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProvider-mount.html">777</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-mount_base.html">94</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-mount.html">88</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">40</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">29</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">71</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/MakeStyles-mount_base.html">872</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/MakeStyles-mount.html">879</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Persona-mount_base.html">1829</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/Persona-mount.html">1755</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/SpinButton-mount_base.html">1426</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/SpinButton-mount.html">1428</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/SwatchPicker-mount_base.html">1721</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32944/perf-test-react-components/SwatchPicker-mount.html">1646</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: apps/vr-tests-react-components/src/stories/Tabs.stories.tsx [hunk 1]
```diff
@@ -206,3 +206,31 @@ export const WithIconOnlyAndVertical = () => (
 );
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Tabs.stories.tsx:206]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32944&commitId=889fbeb0c3c299e170c16fafab4124f4eb93de93&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|7| Changed|
|Avatar Converged.badgeMask.chromium.png|6| Changed|

</details>


  <details><summary><b>TabList and Tab Converged</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|TabList and Tab Converged.Circular Filled Appearance Dark Mode.chromium.png|0| Added|
|TabList and Tab Converged.Circular Filled Appearance High Contrast.chromium.png|0| Added|
|TabList and Tab Converged.Circular Filled Appearance.chromium.png|0| Added|
|TabList and Tab Converged.Circular Subtle Appearance Dark Mode.chromium.png|0| Added|
|TabList and Tab Converged.Circular Subtle Appearance High Contrast.chromium.png|0| Added|
|TabList and Tab Converged.Circular Subtle Appearance.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-tabs/library/etc/react-tabs.api.md [hunk 2]
```diff
@@ -109,11 +110,12 @@ export type TabSlots = {
 
 // @public
 export type TabState = ComponentState<TabInternalSlots> & Pick<TabProps, 'value'> & Required<Pick<TabProps, 'disabled'>> & {
-    appearance?: 'transparent' | 'subtle';
+    appearance?: 'transparent' | 'subtle' | 'filled';
     iconOnly: boolean;
     selected: boolean;
     contentReservedSpaceClassName?: string;
     size: 'small' | 'medium' | 'large';
+    shape?: 'circular';
```
Comment (dmytrokirpa) [inline packages/react-components/react-tabs/library/etc/react-tabs.api.md:118]: I was thinking about two options for this - `circular?: boolean` or `shape?: 'circular'`. Currently, the boolean option appears simpler, but using a discriminated union might enhance scalability and align better with other components like Button, which have shape options. 

I'd appreciate the feedback on this and am willing to change this based on it.

File: packages/react-components/react-tabs/library/src/components/Tab/useTabStyles.styles.ts [hunk 3]
```diff
@@ -173,6 +173,92 @@ const useRootStyles = makeStyles({
...
 
+const useCircularStyles = makeStyles({
+  base: {
+    borderRadius: tokens.borderRadiusCircular,
+    '& .fui-Tab__icon': {
+      color: 'inherit',
+    },
+    '& .fui-Tab__content': {
+      color: 'inherit',
+    },
```
Comment (mainframev) [inline packages/react-components/react-tabs/library/src/components/Tab/useTabStyles.styles.ts:184]:
```suggestion
   [`& ${tabClassNames.icon}`]: {
      color: 'inherit',
    },
   [`& ${tabClassNames.content}`]: {
      color: 'inherit',
    },
```

File: packages/react-components/react-tabs/library/etc/react-tabs.api.md (hunk 2 above)
Comment (mainframev) [reply packages/react-components/react-tabs/library/etc/react-tabs.api.md:118]: makes sense to me, maybe somebody else might have different opinion on this

**Review by mainframev — COMMENTED** [review]
LGTM, few small suggestions

**Review by Hotell — COMMENTED** [review]
the api design is miss-leading ( not referring to your PR -> we need to talk to design ) more context https://github.com/microsoft/fluentui/issues/32951#issuecomment-2386551878


in terms of actual api implementation, we should not introduce another property rather provide discriminant union of variants. any kind of new property addition = hard to maintain property bag

File: packages/react-components/react-tabs/library/src/components/Tab/useTabStyles.styles.ts [hunk 4]
```diff
@@ -522,21 +610,40 @@ export const useTabButtonStyles_unstable = (state: TabState, slot: TabState['roo
...
+    // subtle + circular
+    !disabled && appearance === 'subtle' && isCircular && circularStyles.subtle,
+    !disabled && appearance === 'subtle' && selected && isCircular && circularStyles.subtleSelected,
+    disabled && appearance === 'subtle' && isCircular && circularStyles.subtleDisabled,
+    disabled && appearance === 'subtle' && selected && isCircular && circularStyles.subtleDisabledSelected,
+    // filled + circular
+    !disabled && appearance === 'filled' && isCircular && circularStyles.filled,
+    !disabled && appearance === 'filled' && selected && isCircular && circularStyles.filledSelected,
+    disabled && appearance === 'filled' && isCircular && circularStyles.filledDisabled,
+    disabled && appearance === 'filled' && selected && isCircular && circularStyles.filledDisabledSelected,
```
[the remark this answers is not in the document]
Comment (dmytrokirpa) [reply packages/react-components/react-tabs/library/src/components/Tab/useTabStyles.styles.ts]: updated

**Comment by dmytrokirpa** [comment]
> the api design is miss-leading ( not referring to your PR -> we need to talk to design ) more context [#32951 (comment)](https://github.com/microsoft/fluentui/issues/32951#issuecomment-2386551878)
> 
> in terms of actual api implementation, we should not introduce another property rather provide discriminant union of variants. any kind of new property addition = hard to maintain property bag

I appreciate the feedback, updated PR to use existing prop with new variants.

**Review by mainframev — APPROVED** [review]

**Pull request merged**